main.go

```diff
@@ -1,18 +1,52 @@
 package main
 
 import (
+	"flag"
 	"fmt"
 	"log"
+	"os"
 )
 
 const gomodName = "go.mod"
 
 func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
 
+func run() error {
 	updatedContents, err := MergeRequires(gomodName)
 	if err != nil {
-		log.Fatal(err)
+		return fmt.Errorf("failed to merge requires: %w", err)
+	}
+
+	var inplace bool
+	flag.BoolVar(&inplace, "in-place", false, "replace the contents of go.mod with the updated contents")
+	flag.BoolVar(&inplace, "i", false, "replace the contents of go.mod with the updated contents")
+	flag.Parse()
+
+	// check if we want to replace the contents of go.mod
+	if inplace {
+		return updateInplace(gomodName, updatedContents)
```

Comment:
```suggestion
	if inplace {
		return writeFilePreserveMode(gomodName, updatedContents)
```

Reply:
I think that `updateInplace` explains what the function does, instead of `writeFilePreserveMode` which explain how the function works.

The `run` function cares more that the file is updated in place, instead of the fact the file is written with the mode preserved. 

Inside  `updateInplace` we could find that truncating the file, and then updating with the new contents is more efficient  - and as such the "PreserveMode" not longer really relevant.